addons/resource_mail/models/resource_resource.py

```diff
@@ -8,3 +8,9 @@ class ResourceResource(models.Model):
     _inherit = 'resource.resource'
 
     im_status = fields.Char(related='user_id.im_status')
+
+    def get_avatar_card_data(self, fields):
+        return self.env['resource.resource'].search_read(
+            domain=[('id', '=', self.id)],
```

Comment:
I guess this could work in batch?
```suggestion
            domain=[('id', 'in', self.ids)],
```

Reply:
You're right, the existing implementation wouldn't work correctly in such a case. I don't see any current scenario where this method would be called in batch mode with multiple resources, but to future-proof it, I've updated the code to handle batch processing. 

Thanks for the review!